File: test/normalizers.test.ts
import { describe, expect, it } from 'vitest';

import { filingDocumentUrl } from '../src/sec/endpoints.js';
import { normalizeAccession, normalizeCik, normalizeTicker } from '../src/sec/normalizers.js';

describe('normalizers', () => {
  it('normalizes cik to 10 digits', () => {
    expect(normalizeCik('320193')).toBe('0000320193');
    expect(normalizeCik('0000320193')).toBe('0000320193');
  });

  it('normalizes ticker to uppercase', () => {
    expect(normalizeTicker('aapl')).toBe('AAPL');
  });

  it('normalizes accession format', () => {
    expect(normalizeAccession('0000320193-26-000006')).toBe('0000320193-26-000006');
  });

  it('builds filing url from cik/accession/primary doc', () => {
    const url = filingDocumentUrl({
      cik: '0000320193',
      accession: '0000320193-26-000006',
      primaryDocument: 'aapl-20251227.htm'
    });

    expect(url).toBe(
      'https://www.sec.gov/Archives/edgar/data/320193/000032019326000006/aapl-20251227.htm'
    );
  });
});

File: pkg/edgar/normalizers_test.go
package edgar

import "testing"

func TestNormalizers(t *testing.T) {
	cik, err := normalizeCIK("320193")
	if err != nil {
		t.Fatal(err)
	}
	if cik != "0000320193" {
		t.Fatalf("normalizeCIK mismatch: %q", cik)
	}

	ticker, err := normalizeTicker("aapl")
	if err != nil {
		t.Fatal(err)
	}
	if ticker != "AAPL" {
		t.Fatalf("normalizeTicker mismatch: %q", ticker)
	}

	accession, err := normalizeAccession("0000320193-26-000006")
	if err != nil {
		t.Fatal(err)
	}
	if accession != "0000320193-26-000006" {
		t.Fatalf("normalizeAccession mismatch: %q", accession)
	}
}

func TestFilingDocumentURL(t *testing.T) {
	url, err := filingDocumentURL(defaultSECHosts(), "0000320193", "0000320193-26-000006", "aapl-20251227.htm")
	if err != nil {
		t.Fatal(err)
	}
	want := "https://www.sec.gov/Archives/edgar/data/320193/000032019326000006/aapl-20251227.htm"
	if url != want {
		t.Fatalf("filingDocumentURL = %q, want %q", url, want)
	}
}
